### PR TITLE
fix(web): fix deployment mutability

### PR DESCRIPTION
**Pull request by balanceee**

### CODE REVIEW

File: web/crux-ui/src/components/products/versions/deployments/use-deployment-state.tsx [hunk 1]
```diff
@@ -85,7 +87,9 @@ const useDeploymentState = (options: DeploymentStateOptions): [DeploymentState,
   const [viewMode, setViewMode] = useState<ViewMode>('tile')
   const [confirmationModal, copyDeployment] = useCopyDeploymentModal(onApiError)
 
-  const mutable = deploymentIsMutable(deployment.status)
+  const mutable = deploymentIsMutable(deployment.status, optionDeploy.version.type)
```
Comment (m8vago) [inline web/crux-ui/src/components/products/versions/deployments/use-deployment-state.tsx:90]: just use version.type

File: web/crux-ui/src/components/products/versions/images/edit-image-json.tsx [hunk 2]
```diff
@@ -61,7 +61,9 @@ const EditImageJson = (props: EditImageJsonProps) => {
     (err: Error) => {
       throttle(CANCEL_THROTTLE)
 
-      propOnParseError(err)
+      if (propOnParseError) {
```
Comment (m8vago) [inline web/crux-ui/src/components/products/versions/images/edit-image-json.tsx:64]: is there a use case where it's legit null?

File: web/crux-ui/src/pages/products/[productId]/versions/[versionId]/deployments/[deploymentId]/deploy.tsx [hunk 3]
```diff
@@ -57,7 +57,7 @@ const DeployPage = (props: DeployPageProps) => {
     status,
   }
   const { product, version } = deployment
-  const mutable = deploymentIsMutable(deployment.status)
+  const mutable = deploymentIsMutable(deployment.status, deployment.version.type)
```
Comment (m8vago) [inline web/crux-ui/src/pages/products/[productId]/versions/[versionId]/deployments/[deploymentId]/deploy.tsx:60]: version.type

File: web/crux/src/app/deploy/pipes/deploy.copy.pipe.ts [hunk 4]
```diff
@@ -15,9 +16,23 @@ export default class DeployCopyValidationPipe implements PipeTransform {
...
+        status: true,
+        version: {
+          select: {
+            type: true,
+          },
+        },
       },
     })
 
+    if (!checkDeploymentCopyability(deployment.status, deployment.version.type)) {
```
Comment (m8vago) [inline web/crux/src/app/deploy/pipes/deploy.copy.pipe.ts:28]: typo: copiability

Comment (m8vago) [reply web/crux/src/app/deploy/pipes/deploy.copy.pipe.ts:28]: https://en.wiktionary.org/wiki/copiability#English

File: web/crux/src/domain/deployment.ts [hunk 5]
```diff
@@ -42,22 +42,21 @@ export const containerNameFromImageName = (imageName: string): string => {
   return imageName.substring(index + 1)
 }
 
-export type DeploymentMutabilityCheckDao = {
-  status: DeploymentStatusEnum
-}
+export const checkDeploymentCopyability = (status: DeploymentStatusEnum, type: VersionTypeEnum): boolean =>
```
Comment (m8vago) [inline web/crux/src/domain/deployment.ts:45]: typo

**Review by m8vago — CHANGES_REQUESTED** [review]

File: web/crux-ui/src/components/products/versions/images/edit-image-json.tsx (hunk 2 above)
Comment (balanceee) [reply web/crux-ui/src/components/products/versions/images/edit-image-json.tsx:64]: Yes

Comment (balanceee) [reply web/crux-ui/src/components/products/versions/images/edit-image-json.tsx:64]: But i didn't check its legitability

**Review by m8vago — APPROVED** [review]

File: web/crux/src/shared/mapper.ts [hunk 6]
```diff
@@ -9,3 +10,5 @@ export const toPrismaJson = val => {
 
   return val as JsonObject
 }
+
+export const versionTypeToGrpc = (type: VersionTypeEnum): VersionType => versionTypeFromJSON(type.toUpperCase())
```
Comment (nandor-magyar) [inline web/crux/src/shared/mapper.ts:14]: no NL? :dancing_men:

**Review by nandor-magyar — APPROVED** [review]

**Pull request merged**